### ISSUE TITLE
Possibility to support Application in Resource Conventions

### ISSUE THREAD

**Issue by wdengw** [open-telemetry/opentelemetry-specification#1410]
**What are you trying to achieve?**
Many Java web applications host multiple services. For example, a SpringBoot application can have multiple RestController components, each RestController is a Rest Service. A SpringBoot application usually starts as a JVM process. The OpenTelemetry Log Data model has semantic resource attributes for Service, like service.name, service version, etc. It also has semantic resource attributes for processes, like process.pid, process.command, etc. However, it is lacking the application resource to associate services with the processes. The basic attributes that application resources should have includes: the application’s id, the application’s version, and the application’s group id. 

**What did you expect to see?**
Should we consider to add application as a resource in OpenTelemtry’s Resource Semantic Conventions?
**Additional context.**

**Comment by Oberon00** [comment]
> Many Java web applications host multiple services. 

I think the technology in OTel might be reversed. The service would be the whole running entity that is instrumented with OTel. The application would be something running inside that service. See also #335

**Comment by wdengw** [comment]
@obecny You suggested to consider this springboot application as a server inside HTTP Service? There are a couple of issue with this approach:
1.  The  HTTP Server has no attribute to capture the version, and the groupId of this built springboot artifact;
2. The real problem is from developer's point of view, this springboot app is key entity of their focus for observability.  The use case documented in this issue is multiple services components, it would be multiple other componenets, for example,  message consumers.  The springboot application (or its built deployment artifact with artifactId, groupId, and version) is their deployment unit.  Wrapping this important resources deep inside another HTTP service as HTTP server is not a favorable solution in my opinion.

**Comment by iNikem** [comment]
@wdengw Spring Boot is only of the possible deployment models. If you want to bring the notion of the application into spec, you have to consider other models as well. From the top of my head: 
* Several war files deployed to Java application server like 
* Several hosts defined Apache httpd or Nginx

In general the relation between "process" and "application" is at least "one to many"

**Comment by wdengw** [comment]
@iNikem you are right,  SpringBoot is just one each example, there are many other possible deployment models for developer to deploy their application.  Like what you pointed out, the key is that we deploy applications to deployment environment and applications are what we monitoring.  Therefore, Application is a key resource to be considered.

And yes,  the relation between "process" and "application" is at least "one to many", and in addition,  the relation between "application" and "service" is also at least "one to many".
